libz-rs-sys/src/lib.rs

```diff
@@ -248,7 +226,9 @@ pub extern "C" fn adler32_combine(adler1: c_ulong, adler2: c_ulong, len2: z_off_
 ///
 /// The caller must guarantee that
 ///
-/// * The `destLen` pointer satisfies the requirements of [`core::ptr::read`]
+/// * Either
+///     - `destLen` is `NULL`
+///     - `destLen` satisfies the requirements of `&mut *destLen`
```

Comment:
stock zlib happily dereferences a null pointer here. We check for it being null and return an error, which, because zlib has undefined behavior in this case, is compatible.